README.md

```diff
@@ -33,6 +33,10 @@ $ beanstalkd_exporter -h
 Usage of ./bin/beanstalkd_exporter:
   -beanstalkd.address string
     	Beanstalkd server address (default "localhost:11300")
+  -beanstalkd.connection-retry int
+        A value that allows you to retry to connect beanstalk, if initial connection fails (default 2)
+  -beanstalkd.connection-retry-interval duration
+        Sets a retry interval for beanstalk connection. (default 5s)
```

Comment:
While these address the issue of it failing with no retries, it does not address the issue of it having a connection still hanging.

For that we would need a different PR with a new parameter, maybe `max-beanstalkd-connection-duration`, or a timeout for getting stats `fetch-beanstalkd-stats-timeout`.

As now we have a connection stuck for over a week :), and no logic to handle it.

Reply:
When the prometheus client has called, it will try to create a new connection to beanstalk to collect the stats. If it stucks for one time, in other time it could return error. I will check beanstalk lib, if they are providing connection timeout or if we could add tcp timeout. I will come back to you.

Reply:
From what I've seen (some) connections are left running, they are not closed after stats are received. 
But maybe I'm wrong.

Reply:
I was going to submit a PR for this issue, it seems like we are not closing the connection we open to Beanstalkd. We just need to add a defer close(c) in exporter.go .